media-libs/mesa/mesa-9999.ebuild

```diff
@@ -265,270 +328,218 @@ pkg_setup() {
-		driver_enable video_cards_r200 r200
+		dri_driver_enable video_cards_r100 r100
+		dri_driver_enable video_cards_r200 r200
 		if ! use video_cards_r100 && \
-				! use video_cards_r200; then
-			driver_enable video_cards_radeon radeon r200
+			! use video_cards_r200; then
+			dri_driver_enable video_cards_radeon r100 r200
 		fi
 	fi
 
-	if use egl; then
-		myconf+=" --with-platforms=x11,surfaceless$(use wayland && echo ",wayland")$(use gbm && echo ",drm")"
-	fi
+	emesonargs+=( -Dplatforms=$(use X && echo "x11,")$(use wayland && echo "wayland,")$(use gbm && echo "drm,")surfaceless )
 
 	if use gallium; then
-		myconf+="
-			$(use_enable d3d9 nine)
-			$(use_enable llvm)
-			$(use_enable openmax omx-bellagio)
-			$(use_enable vaapi va)
-			$(use_enable vdpau)
-			$(use_enable xa)
-			$(use_enable xvmc)
-		"
-		use vaapi && myconf+=" --with-va-libdir=/usr/$(get_libdir)/va/drivers"
-
-		gallium_enable swrast
-		gallium_enable video_cards_tegra grate
+		emesonargs+=(
+			$(meson_use llvm)
+			$(meson_use lm-sensors lmsensors)
+			$(meson_use unwind libunwind)
+		)
+
+		if use video_cards_iris ||
+		   use video_cards_r300 ||
+		   use video_cards_r600 ||
+		   use video_cards_radeonsi ||
+		   use video_cards_nouveau ||
+		   use video_cards_vmware; then
+			emesonargs+=($(meson_use d3d9 gallium-nine))
+		else
+			emesonargs+=(-Dgallium-nine=false)
+		fi
+
+		if use video_cards_r600 ||
+		   use video_cards_radeonsi ||
+		   use video_cards_nouveau; then
+			emesonargs+=($(meson_use vaapi gallium-va))
+			use vaapi && emesonargs+=( -Dva-libs-path="${EPREFIX}"/usr/$(get_libdir)/va/drivers )
+		else
+			emesonargs+=(-Dgallium-va=false)
+		fi
+
+		if use video_cards_r300 ||
+		   use video_cards_r600 ||
+		   use video_cards_radeonsi ||
+		   use video_cards_nouveau; then
+			emesonargs+=($(meson_use vdpau gallium-vdpau))
+		else
+			emesonargs+=(-Dgallium-vdpau=false)
+		fi
+
+		if use video_cards_freedreno ||
+		   use video_cards_nouveau ||
+		   use video_cards_vmware; then
+			emesonargs+=($(meson_use xa gallium-xa))
+		else
+			emesonargs+=(-Dgallium-xa=false)
+		fi
+
+		if use video_cards_r600 ||
+		   use video_cards_nouveau; then
+			emesonargs+=($(meson_use xvmc gallium-xvmc))
+		else
+			emesonargs+=(-Dgallium-xvmc=false)
+		fi
+
+		if use video_cards_freedreno ||
+		   use video_cards_lima ||
+		   use video_cards_panfrost ||
+		   use video_cards_vc4 ||
+		   use video_cards_vivante; then
+			gallium_enable -- kmsro
+		fi
+
+		gallium_enable video_cards_lima lima
+		gallium_enable video_cards_panfrost panfrost
 		gallium_enable video_cards_vc4 vc4
 		gallium_enable video_cards_vivante etnaviv
 		gallium_enable video_cards_vmware svga
 		gallium_enable video_cards_nouveau nouveau
-		gallium_enable video_cards_i915 i915
-		gallium_enable video_cards_imx imx
-		if ! use video_cards_i915 && \
-			! use video_cards_i965; then
-			gallium_enable video_cards_intel i915
+		gallium_enable video_cards_tegra tegra
```

Comment:
Hello Kyle,

The `tegra` gallium driver is useful only for T124+. For T20/30 it should be the `grate` gallium driver. We haven't rebased the grate driver on the mesa-20.0, the latest rebase available right now is https://github.com/grate-driver/mesa/tree/19.3. Although, I guess it shouldn't be a problem to add the `grate` option later on